app/services/secret_service.rb

```diff
@@ -2,12 +2,30 @@ class SecretService
 
   # The Secret#secret_key value isn't persisted - it is used to encrypt the secret
   # and the only copy of it is sent in the notification email to the recipient.
-  def self.encrypt_secret(params, request_host)
+  def self.encrypt_existing_secret(id, params, request_host)
```

Comment:
Why aren't all these (`encrypt_existing_secret`, `encrypt_new_secret` and `create_empty_secret`) just methods on the Secret class ?

and the 2 encrypt method seem very redundant.

```
s = Secret.new(params...)
s.encrypt
```
seems a more natural construct.